src/coreclr/jit/morph.cpp

```diff
@@ -11265,8 +11265,10 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                     // If we are performing a narrowing cast and
                     // castType is larger or the same as op1's type
                     // then we can discard the cast.
+                    // However, the actual VM sizes must be the same.
 
-                    if (varTypeIsSmall(castType) && (genTypeSize(castType) >= genTypeSize(effectiveOp1)))
+                    if (varTypeIsSmall(castType) && (genTypeSize(castType) >= genTypeSize(effectiveOp1) &&
+                                                     (genActualTypeSize(castType) == genActualTypeSize(effectiveOp1))))
```

Comment:
I think this needs to compare the input and the output, i.e. `genActualType(castType) == genActualType(op2->AsCast()->CastOp())` (also small nit: fetching the type size should not be necessary for the "actual types").

Reply:
This makes sense, means I can remove the additional function